completes assignment one, version one

scottjason opened the pull request.

scottjason committed:
adds background image and overlay

scottjason committed:
updates remote origin

scottjason committed:
adds stanza logo to navbar

scottjason committed:
adds landing button, adds landing styling to global.css

scottjason committed:
refactors landing css to have one content container with absoutely positioned elems, adds main title

scottjason committed:
completes assignment one minus gradient, last commit meant to say absolte-positioned container with relatively positioned elems

scottjason committed:
adds media queries for tablet and mobile

scottjason committed:
adds comments to landing.css

scottjason committed:
removes unused logo class from navbar logo img

scottjason committed:
cleans up

scottjason committed:
adds nav gradient

scottjason committed:
adds hamburger with transform to add width

scottjason committed:
removes media queries for title copy, adds whitespace no-wrap

scottjason committed:
adds landing line, tests chrome, ff, safari at all widths

scottjason committed:
updates comments and formatting in css

scottjason committed:
removes unsused props out of nav.css

scottjason committed:
updates css animation naming conventions for duration

scottjason committed:
removes unused css props from landing.css

YashYash commented on public/project/css/global.css:
```diff
@@ -1,82 +1,161 @@
 }
 /*Default background*/
+
 #default-background {
```
I am not sure if I added this, or if you did. But can you change this to a class from an id.

scottjason replied:
K .. changed it to a class .. not sure if it's actually doing anything (unless it's in assignments or somewhere else)

YashYash commented on public/project/css/global.css:
```diff
@@ -1,82 +1,161 @@
+  -moz-transition: 0.5;
 }
+
 @keyframes rotating {
```
Can you also delete this rotating keyframe animation. Its not needed. Its something that got copy and pasted in here

YashYash commented on public/project/css/landing.css:
```diff
@@ -0,0 +1,61 @@
+  width: 400px;
+  height: 200px;
+  max-width: 100%;
+}
```
So max-width 100% makes no sense, since 100% is the maximum width lol. So what you can do is, set the width to 100%, and then set the max-width to 400px. It is also reccommended to set the min-width. Usually 280px is the perfect width - because the smallest width you will have to deal with is 320px, which is the width of an iphone. So 280px is the perfect width for mobile.

scottjason replied:
1st sentence .. lol !! k gotcha

YashYash commented on public/project/css/global.css:
```diff
@@ -1,82 +1,161 @@
+  animation-duration: 1s;
+}
+
+.dead-center {
```
Niiiiice

YashYash commented on public/project/css/landing.css:
```diff
@@ -0,0 +1,61 @@
+  text-align: center;
+  margin-top: 14px;
+  white-space: nowrap;
+}
```
Since landing-title is the first element in the container, it is better to set a padding-top on the parent container instead of applying margin-top. Margin is something that is hectic when it comes to working cross browsers, especially pc. So instead of doing margin-top:14px, set padding-top:14px to the parent container. Also, you do not need white-space: nowrap. You just need to set the appropriate min-width to landing-content.

scottjason replied:
k makes sense .. i knew margin was sketch  ie margin-bottom: -50px lol )

YashYash commented on public/project/css/landing.css:
```diff
@@ -0,0 +1,61 @@
+  width: 100px;
+  height: 1px;
+  top: 33px;
+}
```
top:300px would not be applied, since there is no position specified. If you apply position:relative, it will push it down 33px. If you use absolute, it will push it down 33px from the very top

scottjason replied:
the absolute prop comes from the dead center class .. I kinda did the globals as main classes and then for the individuals (ie landing, nav) .. when appropriate .. I'd add their global class and then their "landing-specific-class" would just be "added on" .. so the props in landing-line are being "added on" to the dead-center class which has position absolute. 

Does it matter which class comes first? is this a cool way to go about it or no? Didn't seem like the order mattered since I overrode ..

the top: 0px prop in the "dead-center" class with: 
the top:33px "landing-line" class top:33 

.. even though landing-line is "defined" first in the div's class:

``` html
  <div class="landing-line dead-center"></div>
```

YashYash commented on public/project/css/nav.css:
```diff
@@ -0,0 +1,45 @@
+  background: linear-gradient(to bottom, rgba(0, 0, 0, 0.88) 0%, rgba(5, 5, 5, 0.88) 51%, rgba(10, 10, 10, 0) 99%, rgba(10, 10, 10, 0) 100%);
+  /* IE6-9 */
+  filter: progid: DXImageTransform.Microsoft.gradient( startColorstr='#e0000000', endColorstr='#000a0a0a', GradientType=0);
+}
```
Comments aren't needed. lol

YashYash commented on public/project/css/nav.css:
```diff
@@ -0,0 +1,45 @@
+  top: 24px;
+  transform: scale(1.3, 1);
+}
+
```
When it comes to icons, you want to make it very clickable. When you add padding to an element, it increases its clickable range. So instead of doing top:24px, use top:0 right:0, and then apply padding: 24px 33px;. So now if some one goes to the very top right of the screen, they can still click

scottjason replied:
Cool makes sense .. this element also has a global class (was trying to make re-useable classes) called top-right. So I just replaced top: 24px with padding-top 24px. Will that still cover the same clickable-range?

scottjason committed:
adds yashs fixes, assignment one

scottjason commented:
@YashYash some of the comment-convos are being hidden but you can see em if you click the 'Show outdated diff'

YashYash commented on public/project/css/global.css:
```diff
@@ -1,114 +1,162 @@
+  left: 0;
+  width: 100%;
+  height: 100%;
+  z-index: -10;
```
You should never need to use negative z-index

scottjason replied:
@YashYash That code that came w the repo .. will remove

scottjason committed:
removes default background class that had a negative z index

scottjason committed:
adds gitignore file

scottjason commented:
@YashYash 

I removed that default background class with the negative z index (came with the repo so their might be other weird stuff in there).

Also, I updated the version of mongoose in package.json and added a .gitignore (we shouldn't push the node_modules to github because it takes forever and so that the npm mod versioning is done through the package.json file).

scottjason committed:
updates mongoose in package.json to ~same-version but not latest master
